rbpf-cli/src/main.rs

```diff
@@ -0,0 +1,241 @@
+use clap::{App, Arg};
+use solana_rbpf::{
+    assembler::assemble,
+    ebpf,
+    error::EbpfError,
+    memory_region::{MemoryMapping, MemoryRegion},
+    static_analysis::Analysis,
+    user_error::UserError,
+    verifier::check,
+    vm::{
+        Config, DynamicAnalysis, EbpfVm, Executable, InstructionMeter, SyscallObject,
+        SyscallRegistry,
+    },
+};
+use std::{fs::File, io::Read, path::Path};
+
+pub type Result = std::result::Result<u64, EbpfError<UserError>>;
+
+pub struct TestInstructionMeter {
+    pub remaining: u64,
+}
+
+impl InstructionMeter for TestInstructionMeter {
+    fn consume(&mut self, amount: u64) {
+        debug_assert!(amount <= self.remaining, "Execution count exceeded");
+        self.remaining = self.remaining.saturating_sub(amount);
+    }
+    fn get_remaining(&self) -> u64 {
+        self.remaining
+    }
+}
+
+struct MockSyscall {
+    name: String,
+}
+impl SyscallObject<UserError> for MockSyscall {
+    fn call(
+        &mut self,
+        arg1: u64,
+        arg2: u64,
+        arg3: u64,
+        arg4: u64,
+        arg5: u64,
+        _memory_mapping: &MemoryMapping,
+        result: &mut Result,
+    ) {
+        println!(
+            "Syscall {}: {:#x}, {:#x}, {:#x}, {:#x}, {:#x}",
+            self.name, arg1, arg2, arg3, arg4, arg5,
+        );
+        *result = Result::Ok(0);
+    }
+}
+
+fn main() {
+    let matches = App::new("Solana RBPF CLI")
```

Comment:
Why include the "R" in "RBPF"?  We should probably remove it internally too but at least publicly we should